server/src/test/java/io/crate/integrationtests/ObjectColumnTest.java

```diff
@@ -343,4 +348,30 @@ public void testSelectUnknownObjectColumnPreservesTheUnknownName() throws Except
         // make sure that a['u'] is kept as requested.
         assertThat(printedTable(response.rows())).contains("[(123 = _cast(a['u'], 'integer'))]");
     }
+
+    @Test
+    public void test_default_columns_in_object_result_in_object_creation_if_missing_from_insert() throws Exception {
+        execute("""
+            create table tbl (
+                x int,
+                o object as (
+                    key text default 'default-value'
+                ),
+                os array(object as (
+                    key text default 'default-value'
+                )),
+                complex object as (
+                    os array(object as (
+                        key text default 'nested default'
+                    ))
+                )
+            )
+            """
+        );
+        assertThat(execute("insert into tbl (x) values (1)")).hasRowCount(1);
+        execute("refresh table tbl");
+        assertThat(execute("select o, os, complex from tbl")).hasRows(
+            "{key=default-value}| [{key=default-value}]| {os=[{key=nested default}]}"
```

Comment:
Thinking about this some more I'm not sure if adding a single list item is good. I guess arrays should just be empty?

In 5.2 the case led to an error:

```
cr> create table tbl (
        x int,
        o object as (
            key text default 'default-value'
        ),
        os array(object as (
            key text default 'default-value'
        )),
        complex object as (
            os array(object as (
                key text default 'nested default'
            ))
        )
    );
CREATE OK, 1 row affected  (0.102 sec)

cr> insert into tbl (x) values (1);
INSERT OK, 1 row affected  (0.032 sec)

cr> select * from tbl;
ClassCastException[class java.util.HashMap cannot be cast to class [Ljava.lang.Object; (java.util.HashMap and [Ljava.lang.Object; are in module java.base of loader 'bootstrap')]
```

Reply:
hm true, maybe forbid completely to have defaults for objects within an array?

Reply:
Pushed a fixup to not create arrays